dspace-api/src/main/java/org/dspace/content/EntityTypeServiceImpl.java

```diff
@@ -124,24 +125,33 @@ public void delete(Context context,EntityType entityType) throws SQLException, A
     public List<String> getSubmitAuthorizedTypes(Context context)
             throws SQLException, SolrServerException, IOException {
         List<String> types = new ArrayList<>();
-        StringBuilder query = new StringBuilder();
-        org.dspace.eperson.EPerson currentUser = context.getCurrentUser();
+        StringBuilder query = null;
+        EPerson currentUser = context.getCurrentUser();
         if (!authorizeService.isAdmin(context)) {
             String userId = "";
             if (currentUser != null) {
                 userId = currentUser.getID().toString();
+                query = new StringBuilder();
+                query.append("submit:(e").append(userId);
```

Comment:
It looks like moving the `query.append()` with the `userId` inside the conditional context fixes a bug where the userId _could have_ been null. I don't know how that would happen in a valid context, but the code seems more robust now.